src/algorithm.rs

```diff
@@ -61,3 +61,36 @@ impl From<Algorithm> for String {
         algorithm.to_string()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use serde_json::{from_str, to_string};
+
+    use super::*;
+
+    macro_rules! serde_eq {
+        ($type:ty, $de:literal, $se:expr) => {
+            assert_eq!($de, to_string(&$se).unwrap());
+            assert_eq!(from_str::<$type>($de).unwrap(), $se);
```

Comment:
You can drop the type macro parameter if you help type inference a bit:

```suggestion
            let mut val = $se;
            assert_eq!($de, to_string(&val).unwrap());
            val = from_str($de).unwrap();
            assert_eq!(val, $se);
```

Reply:
Ah! Sweet!